control-plane-operator/hostedclusterconfigoperator/controllers/node/node.go

```diff
@@ -0,0 +1,95 @@
+package node
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	supportutil "github.com/openshift/hypershift/support/util"
+
+	"github.com/openshift/hypershift/support/upsert"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	capiv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	nodePoolAnnotation = "hypershift.openshift.io/nodePool"
+)
+
+type reconciler struct {
+	client             client.Client
+	guestClusterClient client.Client
+	upsert.CreateOrUpdateProvider
+}
+
+func (r *reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+	log.Info("Reconciling")
+
+	node := &corev1.Node{}
+	if err := r.guestClusterClient.Get(ctx, req.NamespacedName, node); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("not found", "request", req.String())
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, fmt.Errorf("failed to get Node: %w", err)
+	}
+
+	var apiErr *apierrors.StatusError
+	nodePoolName, err := r.nodeToNodePoolName(node)
+	if err != nil {
+		if errors.As(err, &apiErr) && !apierrors.IsNotFound(err) {
```

Comment:
s/IsNotFound(err)/IsNotFound(apiErr)/

Reply:
I think is right as it is, isn't it? I want to pass the err I got through apierrors.IsNotFound check.

Reply:
but the check would only work on StatusError no? (otherwise I don't see the point of checking that err contains a StatusError)